test/testother.cpp

```diff
@@ -4472,6 +4472,30 @@ class TestOther : public TestFixture {
               "  }\n"
               "}");
         ASSERT_EQUALS("", errout_str());
+
+        // Ticket #5115 "redundantAssignment when using a union"
+        check("void main(void)\n"
+              "{\n"
+              "    unsigned long lTotal = 0;\n"
+              "    union\n"
+              "    {\n"
+              "        unsigned long l1;\n"
+              "        struct\n"
+              "        {\n"
+              "            unsigned char b1;\n"
+              "            unsigned char b2;\n"
+              "            unsigned char b3;\n"
+              "            unsigned char b4;\n"
+              "        } b;\n"
+              "    } u;\n"
+              "    u.l1 = 1;\n"
+              "    lTotal += u.b.b1;\n"
+              "    u.l1 = 2;\n" //Should not show RedundantAssignment
+              "    lTotal += u.b.b1;\n"
+              "    printf(\"b1 %u, b2 %u, b3 %u b4 %u, l1 %lu, lTotal %lu\",\n"
```

Comment:
I don't see that this printf is needed for this test. maybe the assignment on line 4494 can be remove also?